hack/charts/cluster-api-operator/templates/deployment.yaml

```
@@ -100,9 +97,15 @@ spec:
         - containerPort: 9443
           name: webhook-server
           protocol: TCP
-        - containerPort: {{ ( split ":" $.Values.metricsBindAddr)._1 | int }}
-          name: metrics
+          {{- if $.Values.diagnosticsAddress }}
+          {{- $diagnosticsPort := $.Values.diagnosticsAddress }}
+          {{- if contains ":" $diagnosticsPort -}}
+          {{ $diagnosticsPort = ( split ":" $.Values.diagnosticsAddress)._1 }}
+          {{- end }}
+        - containerPort: {{ $diagnosticsPort | int }}
+          name: diagnostics
```

should this be:
```suggestion
          name: metrics
```

instead of diagnostics? https://github.com/kubernetes-sigs/cluster-api/pull/9264/files#diff-8d4b80dac52bef6a87376c16699a47555f567a9c7b68408965e37b68dedee6cbR33-R35

---

y true, should prop. keep it in sync, even though I am a little confused that they changed the flag name from metrics to diagnostics but kept the metrics name in other places ... I did not look at the changes in dept though :D

---

yeah, they kept in as metrics in CAPI and in other providers so better to keep it aligned IMO

---

Please also re-run `make generate` to have an up-to-date full chart, thanks!

---

Fixed now